### PR TITLE
[IA-4581] update dependencies

### PR DESCRIPTION
**Pull request by mlilynolting**
Routine dependencies update, no need for a version bump :)

Combines branches:
- scala-steward:update/azure-storage-blob-12.24.0
- scala-steward:update/fs2-io-3.9.2
- scala-steward:update/google-cloud-bigquery-2.32.0
- scala-steward:update/google-cloud-billing-2.24.0
- scala-steward:update/google-cloud-compute-1.34.0
- scala-steward:update/google-cloud-container-2.27.0
- scala-steward:update/google-cloud-dataproc-4.21.0
- scala-steward:update/google-cloud-nio-0.127.3
- scala-steward:update/google-cloud-pubsub-1.125.2
- scala-steward:update/google-cloud-resourcemanager-1.26.0
- scala-steward:update/google-cloud-storage-2.27.0
- scala-steward:update/google-cloud-storage-transfer-1.24.0
- scala-steward:update/sbt-dependency-tree-1.9.6
- scala-steward:update/scalafmt-core-3.7.14
- scala-steward:update/swagger-ui-4.19.1

Update changelogs: 

**google-cloud-storage**: 2.24.0 —> 2.27.0
* BlobWriteChannelV2 - same throughput less GC ([#2110](https://github.com/googleapis/java-storage/issues/2110)) ([1b52a10](https://github.com/googleapis/java-storage/commit/1b52a1053130620011515060787bada10c324c0b))
* Update Storage.createFrom(BlobInfo, Path) to have 150% higher throughput ([#2059](https://github.com/googleapis/java-storage/issues/2059)) ([4c2f44e](https://github.com/googleapis/java-storage/commit/4c2f44e28a1ff19ffb2a02e3cefc062a1dd98fdc))
* Update BlobWriteChannelV2 to properly carry forward offset after incremental flush ([#2125](https://github.com/googleapis/java-storage/issues/2125)) ([c099a2f](https://github.com/googleapis/java-storage/commit/c099a2f4f8ea9afa6953270876653916b021fd9f))
* Update GrpcStorageImpl.createFrom(BlobInfo, Path) to use RewindableContent ([#2112](https://github.com/googleapis/java-storage/issues/2112)) ([c805051](https://github.com/googleapis/java-storage/commit/c80505129baa831e492a5514e937875407211595))
* dependency updates
* Implement BufferToDiskThenUpload BlobWriteSessionConfig ([#2139](https://github.com/googleapis/java-storage/issues/2139)) ([4dad2d5](https://github.com/googleapis/java-storage/commit/4dad2d5c3a81eda7190ad4f95316471e7fa30f66))
* Introduce new BlobWriteSession ([#2123](https://github.com/googleapis/java-storage/issues/2123)) ([e0191b5](https://github.com/googleapis/java-storage/commit/e0191b518e50a49fae0691894b50f0c5f33fc6af))
* grpc: Return error if credentials are detected to be null ([#2142](https://github.com/googleapis/java-storage/issues/2142)) ([b61a976](https://github.com/googleapis/java-storage/commit/b61a9764a9d953d2b214edb2b543b8df42fbfa06))
* Possible NPE when HttpStorageOptions deserialized ([#2153](https://github.com/googleapis/java-storage/issues/2153)) ([68ad8e7](https://github.com/googleapis/java-storage/commit/68ad8e7357097e3dd161c2ab5f7a42a060a3702c))
* Update grpc default metadata projection to include acl same as json ([#2150](https://github.com/googleapis/java-storage/issues/2150)) ([330e795](https://github.com/googleapis/java-storage/commit/330e795040592e5df22d44fb5216ad7cf2448e81))
* Make use of ImmutableMap.Builder#buildOrThrow graceful ([#2159](https://github.com/googleapis/java-storage/issues/2159)) ([e9746f8](https://github.com/googleapis/java-storage/commit/e9746f856e9204c1c0ec62f19e6f71ff8a0b9750))
* Update gRPC writeAndClose to only set finish_write on the last message ([#2163](https://github.com/googleapis/java-storage/issues/2163)) ([95df758](https://github.com/googleapis/java-storage/commit/95df758d6753005226556177e68a3e9c630c789b))
* A resumable session without a Range header should be interpreted as 0 length ([#2182](https://github.com/googleapis/java-storage/issues/2182)) ([5302201](https://github.com/googleapis/java-storage/commit/53022011d83e6a8515a5ba008fc45fc2dae39cea))
* Update User-Agent handling for resumable uploads ([#2168](https://github.com/googleapis/java-storage/issues/2168)) ([665b714](https://github.com/googleapis/java-storage/commit/665b714f421d3c13b557d0ff71460c328c010856))
* Update version resolution logic to be more resilient ([#2169](https://github.com/googleapis/java-storage/issues/2169)) ([c89d275](https://github.com/googleapis/java-storage/commit/c89d27508039a014ea5a6dd8d4889f63d07db73f))
* Add new JournalingBlobWriteSessionConfig usable with gRPC transport ([#2194](https://github.com/googleapis/java-storage/issues/2194)) ([8880d94](https://github.com/googleapis/java-storage/commit/8880d94c3d1a737dd4492cf66a16ba5e08633a70))
* Follow-up CLI Improvements ([#2184](https://github.com/googleapis/java-storage/issues/2184)) ([d985976](https://github.com/googleapis/java-storage/commit/d9859768081ea6f872097851d3e318b5bad384d9))
* Initial CLI for SSB integration and Workload 1 ([#2166](https://github.com/googleapis/java-storage/issues/2166)) ([a349735](https://github.com/googleapis/java-storage/commit/a349735e7fe108e623a330afec0c8cd608ebeef9))

**google-cloud-nio**: 0.127.2 —> 1.127.3
* Update actions/checkout action to v4 ([#1245](https://github.com/googleapis/java-storage-nio/issues/1245)) ([9d0a175](https://github.com/googleapis/java-storage-nio/commit/9d0a17581020058a35c592dce895a7e601faf584))
* dependency updates

**google-cloud-pubsub**: 1.124.2 —> 1.125.2
* Make retry policy back off more aggressively for RPCs that retry RESOURCE_EXHAUSTD ([#1704](https://github.com/googleapis/java-pubsub/issues/1704)) ([f61e7e0](https://github.com/googleapis/java-pubsub/commit/f61e7e0e2895a90e30ecf4dcbfa8c70022947d9e))
* Update actions/checkout action to v4
* dependency updates

**google-cloud-kms**: 2.26.0 —> 2.27.0
* minor formatting
* Update dependency com.google.cloud:google-cloud-shared-dependencies to v3.15.0 ([#9809](https://github.com/googleapis/google-cloud-java/issues/9809)) ([09dbd85](https://github.com/googleapis/google-cloud-java/commit/09dbd855f683b40a462c4f918511bee4671e0174))

**google-cloud-compute**: 1.33.0 —> 1.34.0
* update dependency com.google.cloud:google-cloud-shared-dependencies to v3.15.0

**google-cloud-dataproc**: 4.20.0 —> 4.21.0
* support min_num_instances for primary worker and InstanceFlexibilityPolicy for secondary worker
* update dependency com.google.cloud:google-cloud-shared-dependencies to v3.15.0

**google-cloud-storage-transfer**: 1.23.0 —> 1.24.0
* update dependency com.google.cloud:google-cloud-shared-dependencies to v3.15.0

**google-cloud-resourcemanager**: 1.25.0 —> 1.26.0
* update dependency com.google.cloud:google-cloud-shared-dependencies to v3.15.0

**google-cloud-bigquery:**  2.31.2  —> 2.32.0
* Add support for converting interval fields to threeten PeriodDuration ([#2838](https://github.com/googleapis/java-bigquery/issues/2838)) ([2294c2f](https://github.com/googleapis/java-bigquery/commit/2294c2ffca62a22a66786a9a4c6c9ef1be898e5d))
* Add support for ExternalDatasetReference ([#2871](https://github.com/googleapis/java-bigquery/issues/2871)) ([bbb86fd](https://github.com/googleapis/java-bigquery/commit/bbb86fd8488ad253f2e9cf3fb08360330bd860a3))

**google-cloud-billing**: 2.23.0  —> 2.24.0
* fixed resource_reference for name in GetProjectBillingInfo

**fs2-io**: 3.9.1 —> 3.9.2
* Add Channel#closeWithElement by [@tothpeti](https://github.com/tothpeti) in [#3280](https://github.com/typelevel/fs2/pull/3280)
* Remove @nowarn213 annotation from io package object by [@armanbilge](https://github.com/armanbilge) in [a9566bd](https://github.com/typelevel/fs2/commit/a9566bdf2a7ca506c42df375a17f79b3aa69dda7)
* dependency updates

**azure-storage-blob**: 12.23.1 —> 12.24.0
* Added support for service versions 2023-05-03 and 2023-08-03.
* Added RehydratePendingToCold value to ArchiveStatus enum.

**PR checklist**
- [x] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [x] Bump the version in `project/Settings.scala` `createVersion()`
- [x] Update `CHANGELOG.md` for those libraries
- [x] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [x] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

File: google/CHANGELOG.md [hunk 1]
```diff
@@ -4,13 +4,12 @@ This file documents changes to the `workbench-google` library, including notes o
...
 ## 0.29
 
-SBT Dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-google" % "0.29-7362eef"`
+SBT Dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-google" % "0.TRAVIS-REPLACE-ME"`
 
 ### Dependency upgrades
 | Dependency   |     Old Version      |          New Version |
 |----------|:--------------------:|---------------------:|
 | cats-effect |        3.4.11        |                3.5.1 |
-| fs2-io |        3.6.1         |                3.8.0 |
```
Comment (lucymcnatt) [inline google/CHANGELOG.md:13]: not in the google dependencies list on dependencies.scala, so I removed it

File: project/build.properties [hunk 2]
```diff
@@ -1 +1 @@
-sbt.version=1.8.2
+sbt.version=1.9.6
```
Comment (mlilynolting) [inline project/build.properties:1]: Leo is already on this version of sbt, so this is hopefully safe.

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 addSbtPlugin("org.scoverage" % "sbt-scoverage" % "2.0.9")
 
-addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.5.2")
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "3.7.14")
```
Comment (mlilynolting) [inline project/plugins.sbt]: This is erroneous and I will revert.

**Review by LizBaldo — COMMENTED** [review]
Thanks for doing this, can you also update the Readme with `TRAVIS-REPLACE-ME`?

**Comment by mlilynolting** [comment]
> Thanks for doing this, can you also update the Readme with `TRAVIS-REPLACE-ME`?

oof, good catch! done.

**Review by LizBaldo — APPROVED** [review]

**Pull request merged**